### PR TITLE
Allow an additional handler to be specified for uncaught exceptions.

### PR DESCRIPTION
**Pull request by technomancy**
This is needed to allow the new version of rollcage to be used here:
https://github.com/circleci/service-utils/pull/110

### CODE REVIEW

File: src/circleci/rollcage/core.clj [hunk 1]
```diff
@@ -255,13 +255,15 @@
...
+  "Setup handler to report all uncaught exceptions to rollbar, and optionally
+  to an additional handler."
+  ([client] (setup-uncaught-exception-handler client identity))
+  ([client handler]
+   (Thread/setDefaultUncaughtExceptionHandler
     (reify Thread$UncaughtExceptionHandler
       (uncaughtException [_ thread ex]
-        (report-uncaught-exception "critical" client ex thread)))))
+        (report-uncaught-exception "critical" client ex thread)
+        (handler ex))))))
```
Comment (gordonsyme) [inline src/circleci/rollcage/core.clj]: `handler` needs to have the signature `[thread ex]` to be generally useful.

Comment (technomancy) [reply src/circleci/rollcage/core.clj]: Good call; thanks.

**Comment by technomancy** [comment]
Does that look better?

**Review by dgoeke — APPROVED** [review]
LGTM

**Comment by technomancy** [comment]
Thanks.

**Comment by JustinC474** [comment]
@technomancy should we add a test for the failure case? seems simple enough and this repo is almost at 100% coverage

**Pull request merged**